app/src/main/java/com/example/whatsapp_clone/Views/Fragments/LoginViewModel.java

```diff
@@ -63,50 +41,53 @@ public LiveData<Result.Error> getLoginError() {
      * @param password The user's password.
      */
     public void loginUser(String username, String password) {
-        // Connect to Firebase
-//        auth = FirebaseAuth.getInstance();
-//        auth.signInWithEmailAndPassword(username, password);
-//        User user = new User(username, password, "picture");
-
         // Login request with username and password
-        loginRepository.handleLogin(username, password, new LoginRepository.LoginCallback() {
-            public void onSuccess(String token) {
-                // Get the user
-                loggedInUser.postValue(user);
-                saveUserPreferences(loggedInUser.username, loggedInUser.displayName, loggedInUser.profilePic, token);
-            }
-
-            public void onError(Result.Error error) {
-                // Fail login message
-//                Toast.makeText(getContext(), "Something went wrong!", Toast.LENGTH_LONG).show();
-                // Handle login error
-                loginError.setValue(error);
+        repository.handleLogin(username, password, new CompletionBlock<Token>() {
+            @Override
+            public void onResult(Result<Token> result) {
+                // Check if result is an error
+                if (result.isSuccess()) {
+                    // Get the user and send it to the chat screen
+                    handleUser(username, result.getData());
+                } else {
+                    // Handle the error message
+                    String errorMsg = result.getErrorMessage();
+                    loginError.setValue(new Result<String>(false, "ERROR", errorMsg));
+                }
             }
         });
+//        repository.handleLogin(username, password, new Callback<Token>() {
+//            public void onSuccess(String token) {
+//                // Get the user and send it to the chat screen
+//                handleUser(username, token);
+//            }
+//
+//            public void onError(Result error) {
+//                // Handle login error
+//                loginError.setValue(error);
+//            }
+//        });
     }
 
     /**
      * Get the user profile for the specified user ID.
-     * @param userId The user ID.
+     * @param username The user ID.
      */
-    private void getUserProfile(String userId) {
-        // Make an API call to get the user profile
-        Call<UserProfile> call = apiService.getUserProfile(userId);
-        call.enqueue(new Callback<UserProfile>() {
+    private void handleUser(String username, Object token) {
+        // Request user profile from the repository
+        repository.getUser(username, (String) token, new CompletionBlock<User>() {
```

Comment:
this wont work since you pass a Token object to this function, casting should fail